### PR TITLE
Fix glossary header bug

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Guide pages that display a `summary-box` component also display the **Glossary** `h2` when it should not be visible.
`glossary.html` is unnecessarily rendered on guide pages that don't use a glossary.

See [guides/dap/common-questions-about-dap](https://digital.gov/guides/dap/common-questions-about-dap/)
Screen Shot 2024-03-01 at 2 08 39 PM

### Preview
[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/dap/common-questions-about-dap/)

### Solution
1. Add a conditional check to display `glossary.html` when `.Parent.Params.glossary` is true.
2. If the `summary-box` component is used on a page with a `glossary` component enabled, then don't display the **Glossary** `h2`.
3. Hide the **Glossary** `h2` in `summary-box.js`

### How To Test

1. Visit [guide page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/dap/common-questions-about-dap/) and should not see **glossary** header in `summary box` when glossary **is not** enabled.
2. Visit [guide page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/hcd/discovery-concepts/present/) and should not see **glossary** header in `summary box` when glossary **is** enabled.
	- set `summary_box: true` on `content/guides/hcd/discovery-concepts/present.md` on your local machine
2. Visit [guide landing page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-fix-guide-glossary-bug/guides/hcd/introduction/) and should not see **glossary** header in `summary box` when glossary **is** enabled.
	- set `summary_box: true` on `content/guides/hcd/introduction/_index.md` on your local machine

### What To Test

- [ ] `glossary.html` is not rendered when `glossary` field is not set
	- `dg-glossary__container` is not present in markup
- [ ] `summary-box` component does not display **Glossary** `h2` at all (even when set).

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/layouts/guides/single.html [hunk 1]
```diff
@@ -19,7 +19,9 @@
       {{- else -}}
         {{- partial "core/guides/guide-content.html" . -}}
       {{- end -}}
-      {{- partial "core/glossary.html" . -}}
+      {{- if .Parent.Params.glossary -}}
+        {{- partial "core/glossary.html" . -}}
+      {{- end -}}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/guides/single.html:24]: Check if `.Parent` page has `glossary` field set. This refers to the root `_index.md` in the guide directory.

File: themes/digital.gov/layouts/partials/core/glossary.html [hunk 2]
```diff
@@ -13,7 +13,7 @@
     </svg>
   </button>
 
-  <h2>Glossary</h2>
+  <h2 class="dg-glossary__header">Glossary</h2>
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/glossary.html:16]: Added class as a hook for the javascript.

File: themes/digital.gov/src/js/summary-box.js [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
   const guideSummaryList = guideSummary.querySelector(".usa-list");
   const pageHeaders = document.querySelectorAll(
-    "h2:not(.usa-summary-box__heading, .dg-guide__content-header-title)"
+    "h2:not(.usa-summary-box__heading, .dg-guide__content-header-title, .dg-glossary__header)"
```
Comment (nick-mon1) [inline themes/digital.gov/src/js/summary-box.js:13]: This query acts as "blacklist" of headers to not in the `summary box` component depending on the page it is called from.

**Comment by nick-mon1** [comment]
[DG Bug - remove glossary H2 when not enabled on regular guide pages](https://trello.com/c/p74qol0k/825-dg-bug-remove-glossary-h2-when-not-enabled-on-regular-guide-pages)

**Review by clmedders — APPROVED** [review]
LGTM!

File: themes/digital.gov/layouts/partials/core/glossary.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/glossary.html:1]: Can we add a comment for this component?

It'd be nice to add what it is, where/how its used, and dependencies like `summary-box.js`.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Works well, nice job. Minor comments on adding documentation.

File: themes/digital.gov/layouts/partials/core/glossary.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/glossary.html:1]: Yes, good call @mejiaj

File: themes/digital.gov/src/js/summary-box.js [hunk 4]
```diff
@@ -10,7 +10,7 @@
...
   );
 
   function createSummaryBox() {
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/summary-box.js:24]: JSDocs added 👍🏻

**Review by mejiaj — CHANGES_REQUESTED** [review]
Looks good overall. Added suggestions for clarity & code comments.

File: themes/digital.gov/src/js/summary-box.js [hunk 5]
```diff
@@ -1,18 +1,26 @@
-// Guide Table of Contents
-// Replaces guidenav.js that used Hugo's .TableOfContents with Summary Box component — https://designsystem.digital.gov/components/summary-box/#package
-// Get all H2's and create a top of the page Table of Contents if shortcode is used on the markdown page
-
+/**
+ * Guide Table of Contents
+ * Replaces guidenav.js that used Hugo's .TableOfContents with USWDS summary-box — https://designsystem.digital.gov/components/summary-box/#package
+ * Get all H2's and create a top of the page Table of Contents if shortcode is used on the markdown page
+ */
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/summary-box.js:5]: Added a link to example page and markdown file and added more detail on how this component is implemented

**Review by mejiaj — APPROVED** [review]
LGTM, removed duplicate header in c993905b15.

**Review by clmedders — APPROVED** [review]

**Pull request merged**